### ISSUE TITLE
Typematic rate setting

### ISSUE THREAD

**Issue by brainstorm** [lambdalisue/jupyter-vim-binding#35]
Is there an easy way via config to set the "repeat rate", (or typematic rate)? That is the amount of times the notebook will repeat a single character when its key is held down.

Thanks in advance!

**Comment by lambdalisue** [comment]
I think it is in your OS.

**Comment by lambdalisue** [comment]
Let me know the detail. If you mean

### Scroll via j/k in Jupyter's command mode

https://github.com/lambdalisue/jupyter-vim-binding#customize

### Scroll via C-y/C-d in Vim's command mode

I'm not sure but you may be possible to define your custom map like

``` javascript
CodeMirror.Vim.map('<C-y>', '2<C-y>', 'normal');
```

See https://github.com/lambdalisue/jupyter-vim-binding#customize

### Speed when you hold the key in anywhere in your OS.

Depends on OS.
Windows: http://windows.microsoft.com/en-us/windows-xp/help/adjust-the-character-repeat-rate
MacOSX: http://apple.stackexchange.com/questions/10467/how-to-increase-keyboard-key-repeat-rate-on-os-x
Ubuntu: http://askubuntu.com/questions/140255/how-to-override-the-new-limited-keyboard-repeat-rate-limit

I googled with "speed key repeat"

**Comment by brainstorm** [comment]
Ouch, yes, you were right, it was just a settings issue... I just ran the `defaults` command, thank you! ;)

**Comment by brainstorm** [comment]
Wait, tried those changes... most bizarre thing is that arrow keys do support repeat while the rest of keys do not :/

**Comment by lambdalisue** [comment]
So, can you explain how can i reproduce the issue?

**Comment by brainstorm** [comment]
Yes, on OSX Yosemite, you setup the key-repeating parameters above, then:

1) Try to hold down and repeat a random character in either INSERT or commands mode, i.e: eeeeee
2) Hold down arrow keys in any of the modes.

In case 1, keys do not get repeated, in case 2, they do since the cursor moves around. But then again, as you pointed out, it is an OS issue since for instance in a text editor or browser's navigation bar it behaves like so.

I suspect that having several keyboards as I do activates the alternate chars feature in OSX:

![skarmavbild 2015-12-16 kl 11 24 24](https://cloud.githubusercontent.com/assets/175587/11838492/b5530232-a3e7-11e5-88f9-a2d1a21ea9e5.png)

Perhaps then OSX blocks key repetition for all keys, even when they don't have alternative representations. Really silly issue, sorry to make you spend time on this one, I'm sure it's a silly setting to flip and all is fixed, but haven't found it yet :/

**Comment by lambdalisue** [comment]
> as you pointed out, it is an OS issue since for instance in a text editor or browser's navigation bar it behaves like so.
> I suspect that having several keyboards as I do activates the alternate chars feature in OSX

Probably. I was using OSX Yosemite and am using El capitan but never be able to reproduce this issue.

I'll close this but feel free to re-open if you found that this is actually an issue of the plugin.

**Comment by aswolf** [comment]
I am actually getting the same problem as described here. I am running mac OS 10.9.5.  If i hold down letter keys in insert mode, it activates the alternate character menu to pop up:
<img width="711" alt="screenshot 2016-09-28 09 18 22" src="https://cloud.githubusercontent.com/assets/2340371/18914841/b0d5a564-855c-11e6-94f2-985dff0ed409.png">

I do not think that this is an OS issue, as I have used this same computer/setup for 3 years without any problems. As far as I can tell, this behavior only appears in jupyter-vim-binding.

Your extension is awesome and I would love to use it. Any help is greatly appreciated!

**Comment by aswolf** [comment]
Scratch that! It was OS X. I just never noticed this behavior. You show how to fix it in #40 . For anyone else, issue this command on a terminal and restart web browser:

```
defaults write -g ApplePressAndHoldEnabled -bool false
```

**Comment by lambdalisue** [comment]
Grad to hear :smile:
